### PR TITLE
feat: [CCM-19101]: CCM Plugin for IDP

### PR DESCRIPTION
**Pull request by rajarshee-harness**
## Describe your changes
This PR adds a CCM plugin for Backstage. 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.

### CODE REVIEW

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 1]
```diff
@@ -222,6 +228,32 @@ const chaosContent = (
...
+  <EntitySwitch>
+    <EntitySwitch.Case if={isHarnessCcmAvailable}>
+      <EntityCcmContent />
+    </EntitySwitch.Case>
+
+    <EntitySwitch.Case>
+      <EmptyState
+        title="No Cloud Cost Management data available for this entity"
+        missing="info"
+        description="You need to add an annotation to your component if you want to enable Cloud Cost Management for it. You can read more about annotations in Backstage by clicking the button below."
```
Comment (OrkoHunter) [inline packages/app/src/components/catalog/EntityPage.tsx:242]: Can we add the annotation info here? `harness.io/perspective-url`

**Review by OrkoHunter — COMMENTED** [review]
Great work Rajarshee!

Please ensure you update the main README.md as well marking CCM as done and pointing to the installation instructions.

We also need a README inside the plugin with couple of screenshots. For example - https://github.com/harness/backstage-plugins/tree/main/plugins/harness-iacm

File: packages/app/src/components/catalog/EntityPage.tsx (hunk 1 above)
Comment (rajarshee-harness) [reply packages/app/src/components/catalog/EntityPage.tsx]: Thanks for the feedback! I have made the requested changes.

**Review by Debanitrkl — DISMISSED** [review]
LGTM

**Comment by OrkoHunter** [comment]
@Debanitrkl @rajarshee-harness  Let's ensure it works with
1. vanity urls (app3.harness.io, company.harness.io, etc.)
2. old and new nav urls both

We can't merge before that as it leads to confusion after deployment.

File: plugins/harness-ccm/src/hooks/useResourceSlugFromEntity.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { match } from 'path-to-regexp';
+
+export const useResourceSlugFromEntity = (perspectiveUrl?: string) => {
+  const cleanedString = perspectiveUrl?.split(' ')[0];
+  let accountId;
+  let orgId;
+  let projectId;
+  let perspectiveId;
+
+  const urlMatch = match(
```
Comment (Debanitrkl) [inline plugins/harness-ccm/src/hooks/useResourceSlugFromEntity.tsx:10]: @rajarshee-harness here we need to add all the URLs that are available for perspective ID, like new nav old nav unified view, example https://github.com/harness/backstage-plugins/blob/41c06019d6ce9c3991322956c766342103db8354/plugins/harness-ci-cd/src/hooks/useProjectSlugEntity.tsx#L25-L67

**Review by Debanitrkl — APPROVED** [review]
LGTM 🚀

**Pull request merged**